src/activity_item_handling.cpp

```diff
@@ -640,3 +648,98 @@ void activity_on_turn_move_items()
         }
     }
 }
+
+cata::optional<tripoint> find_best_fire( const std::vector<tripoint> &from, const tripoint &center )
+{
+    cata::optional<tripoint> best_fire;
+    time_duration best_fire_age = 1_days;
+    for( const tripoint &pt : from ) {
+        field_entry *fire = g->m.get_field( pt, fd_fire );
+        if( fire == nullptr || fire->getFieldDensity() > 1 ||
+            !g->m.clear_path( center, pt, PICKUP_RANGE, 1, 100 ) ) {
+            continue;
+        }
+        time_duration fire_age = fire->getFieldAge();
+        // Refuel only the best fueled fire (if it needs it)
+        if( fire_age < best_fire_age ) {
+            best_fire = pt;
+            best_fire_age = fire_age;
+        }
+        // If a contained fire exists, ignore any other fires
```

Comment:
I forgot to change it after the range increase, though it's next to trivial.
I'll leave it as is for now since it's tested already. It's pretty unlikely that a player will have two contained fires anyway.